src/main/kotlin/org/koitharu/kotatsu/parsers/site/HoneyMangaParser.kt

```diff
@@ -0,0 +1,206 @@
+					url = jo.optString("chapterResourcesId"),
+					scanlator = null,
+					uploadDate = dateFormat.tryParse(jo.getString("lastUpdated")),
+					branch = null,
+					source = source
+				)
+			}
+		)
+	}
+
+	override suspend fun getListPage(
+		page: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder
+	): List<Manga> {
+		val body = JSONObject()
+		var content: JSONArray? = null
+		content = if (!query.isNullOrEmpty()) {
+			if((query.length < 3) || (page > 1)) return emptyList()
+			body.put("query", query)
+			webClient.httpGet(searchApi + query).parseJsonArray()
+		} else {
+			body.put("page", page)
+			body.put("pageSize", PAGE_SIZE)
+			body.put("filters", JSONArray())
+			val sort = JSONObject()
+			sort.put("sortBy", getSortKey(sortOrder))
+			sort.put("sortOrder", "DESC")
+			body.put("sort", sort)
+			webClient.httpPost(mangaApi, body).parseJson().getJSONArray("data")
+		}
+		return content!!.mapJSON { jo ->
+			val id = jo.getString("id")
+			val posterUrl = jo.getString("posterUrl")
+			Manga(
+				id = generateUid(id),
+				title = jo.getString("title"),
+				altTitle = jo.getStringOrNull("alternativeTitle"),
+				url = id,
+				publicUrl = "https://$domain/book/$id",
+				rating = RATING_UNKNOWN,
+				isNsfw = isNsfw(jo.getStringOrNull("adult")),
+				coverUrl = getCoverUrl(posterUrl, 256),
+				tags = getTags(jo.optJSONArray("genresAndTags")!!),
+				state = when (jo.getStringOrNull("titleStatus")) {
+					"Онгоінг" -> MangaState.ONGOING
+					"Завершено" -> MangaState.FINISHED
+					else -> null
+				},
+				author = null,
+				largeCoverUrl = getCoverUrl(posterUrl, 1080),
+				description = jo.getStringOrNull("description"),
+				chapters = null,
+				source = source,
+			)
+		}
+	}
+
+	override suspend fun getPages(chapter: MangaChapter): List<MangaPage> {
+		val body = JSONObject()
+		body.put("chapterId", chapter.url)
+
+		val content = webClient.httpPost(framesApi, body).parseJson().getJSONObject("resourceIds")
+		val mangaPage = mutableListOf<MangaPage>()
+		(0 until content.length()).forEach { i ->
+			val item = content.get(i.toString()).toString()
+
+			mangaPage.add(MangaPage(id = generateUid(item), "$imageStorageUrl/$item", getCoverUrl(item, 256), source))
+		}
+		return mangaPage
+	}
+
+	override suspend fun getTags(): Set<MangaTag> {
+		// https://data.api.honey-manga.com.ua/genres-tags/genres-list
+		val tagsSet = mutableListOf<MangaTag>()
+		val content = webClient.httpGet(genresListApi).parseJsonArray()
+		(0 until content.length()).forEach { i ->
+			val item = content.get(i).toString()
+
+			tagsSet.add(MangaTag(item, item, source))
+		}
+
+		return tagsSet.toSet()
+	}
+
+	// Need for disable encoding (with encoding not working)
+	override fun intercept(chain: Interceptor.Chain): Response {
+		val request = chain.request()
+		val newRequest = if (request.header("Content-Encoding") != null) {
+			request.newBuilder().removeHeader("CommonHeaders.CONTENT_ENCODING").build()
```

Comment:
```suggestion
			request.newBuilder().removeHeader("Content-Encoding").build()
```

Or move it to the `const val`